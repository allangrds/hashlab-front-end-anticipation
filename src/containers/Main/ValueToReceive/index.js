import React from 'react'
import PropTypes from 'prop-types'

import centsToReal from '../../../utils/helpers/centsToReal'
import styles from './styles.module.css'

const ValueToReceive = ({
  in15Days,
  in30Days,
  in90Days,
  tomorrow,
}) => (
  <aside className={styles.aside}>
    <p className={styles.title}>
      Você receberá
    </p>
    <hr className={styles.line} />
    {
      tomorrow && (
        <p className={styles.valueDay}>
          Amanhã: <b>{centsToReal(tomorrow)}</b>
        </p>
      )
    }
    {
      in15Days && (
        <p className={styles.valueDay}>
          Em 15 dias: <b>{centsToReal(in15Days)}</b>
        </p>
      )
    }
    {
      in30Days && (
        <p className={styles.valueDay}>
          Em 30 dias: <b>{centsToReal(in30Days)}</b>
        </p>
      )
    }
    {
      in90Days && (
        <p className={styles.valueDay}>
          Em 90 dias: <b>{centsToReal(in90Days)}</b>
        </p>
      )
    }
  </aside>
)

ValueToReceive.propTypes = {
  in15Days: PropTypes.number,
  in30Days: PropTypes.number,
  in90Days: PropTypes.number,
  tomorrow: PropTypes.number,
}

ValueToReceive.defaultProps = {
  in15Days: null,
  in30Days: null,
  in90Days: null,
  tomorrow: null,
}

export default ValueToReceive
