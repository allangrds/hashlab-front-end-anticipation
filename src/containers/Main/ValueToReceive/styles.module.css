@import "../../../assets/css/settings/colors.css";
@import "../../../assets/css/tools/fonts.css";

.aside {
  background-color: var(--color-gray-5);
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 20px;
  height: 100%;
}

.title {
  color: var(--color-blue-1);
  font-family: "Source Sans Pro", sans-serif;
  font-size: 16px;
  font-weight: bold;
  margin-bottom: 4px;
  margin-top: 0;
  text-transform: uppercase;
}

.line {
  border: 1px solid var(--color-blue-2);
  width: 100%;
  max-width: 161px;
  margin-top: 0;
  margin-left: 0;
  margin-right: 0;
  margin-bottom: 23px;
  text-align: left;
}

.valueDay {
  color: var(--color-blue-2);
  font-family: "Source Sans Pro", sans-serif;
  font-style: italic;
  font-size: 16px;
  margin-top: 0;
}
