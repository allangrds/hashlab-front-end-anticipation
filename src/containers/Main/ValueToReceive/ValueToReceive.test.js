import React from 'react'
import { render } from '@testing-library/react'

import ValueToReceive from '.'

test('ValueToReceive should render correctly', () => {
  const in15Days = 1100
  const in30Days = 1200
  const in90Days = 1300
  const tomorrow = 1000

  const formatedIn15Days = 'R$ 11,00'
  const formatedIn30Days = 'R$ 12,00'
  const formatedIn90Days = 'R$ 13,00'
  const formatedTomorrow = 'R$ 10,00'

  const {
    container,
    getByText,
  } = render(
    <ValueToReceive
      in15Days={in15Days}
      in30Days={in30Days}
      in90Days={in90Days}
      tomorrow={tomorrow}
    />,
  )

  getByText(formatedIn15Days)
  getByText(formatedIn30Days)
  getByText(formatedIn90Days)
  getByText(formatedTomorrow)
  expect(container).toMatchSnapshot()
})

test('ValueToReceive should render correctly without tomorrow', () => {
  const in15Days = 1100
  const in30Days = 1200
  const in90Days = 1300
  const tomorrow = null

  const formatedIn15Days = 'R$ 11,00'
  const formatedIn30Days = 'R$ 12,00'
  const formatedIn90Days = 'R$ 13,00'

  const {
    container,
    getByText,
    queryByText,
  } = render(
    <ValueToReceive
      in15Days={in15Days}
      in30Days={in30Days}
      in90Days={in90Days}
      tomorrow={tomorrow}
    />,
  )

  getByText(formatedIn15Days)
  getByText(formatedIn30Days)
  getByText(formatedIn90Days)
  expect(queryByText('AmanhÃ£:')).not.toBeInTheDocument()

  expect(container).toMatchSnapshot()
})

test('ValueToReceive should render correctly without 15 days', () => {
  const in15Days = null
  const in30Days = 1200
  const in90Days = 1300
  const tomorrow = 1000

  const formatedIn30Days = 'R$ 12,00'
  const formatedIn90Days = 'R$ 13,00'
  const formatedTomorrow = 'R$ 10,00'

  const {
    container,
    getByText,
    queryByText,
  } = render(
    <ValueToReceive
      in15Days={in15Days}
      in30Days={in30Days}
      in90Days={in90Days}
      tomorrow={tomorrow}
    />,
  )

  getByText(formatedTomorrow)
  getByText(formatedIn30Days)
  getByText(formatedIn90Days)
  expect(queryByText('Em 15 dias:')).not.toBeInTheDocument()

  expect(container).toMatchSnapshot()
})

test('ValueToReceive should render correctly without 30 days', () => {
  const in15Days = 1100
  const in30Days = null
  const in90Days = 1300
  const tomorrow = 1000

  const formatedIn15Days = 'R$ 11,00'
  const formatedIn90Days = 'R$ 13,00'
  const formatedTomorrow = 'R$ 10,00'

  const {
    container,
    getByText,
    queryByText,
  } = render(
    <ValueToReceive
      in15Days={in15Days}
      in30Days={in30Days}
      in90Days={in90Days}
      tomorrow={tomorrow}
    />,
  )

  getByText(formatedTomorrow)
  getByText(formatedIn15Days)
  getByText(formatedIn90Days)
  expect(queryByText('Em 30 dias:')).not.toBeInTheDocument()

  expect(container).toMatchSnapshot()
})

test('ValueToReceive should render correctly without 90 days', () => {
  const in15Days = 1100
  const in30Days = 1200
  const in90Days = null
  const tomorrow = 1000

  const formatedIn15Days = 'R$ 11,00'
  const formatedIn30Days = 'R$ 12,00'
  const formatedTomorrow = 'R$ 10,00'

  const {
    container,
    getByText,
    queryByText,
  } = render(
    <ValueToReceive
      in15Days={in15Days}
      in30Days={in30Days}
      in90Days={in90Days}
      tomorrow={tomorrow}
    />,
  )

  getByText(formatedTomorrow)
  getByText(formatedIn15Days)
  getByText(formatedIn30Days)
  expect(queryByText('Em 90 dias:')).not.toBeInTheDocument()

  expect(container).toMatchSnapshot()
})
