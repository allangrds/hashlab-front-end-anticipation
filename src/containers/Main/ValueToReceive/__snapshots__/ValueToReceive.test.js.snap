// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValueToReceive should render correctly 1`] = `
<div>
  <aside
    class="aside"
  >
    <p
      class="title"
    >
      Você receberá
    </p>
    <hr
      class="line"
    />
    <p
      class="valueDay"
    >
      Amanhã: 
      <b>
        R$ 10,00
      </b>
    </p>
    <p
      class="valueDay"
    >
      Em 15 dias: 
      <b>
        R$ 11,00
      </b>
    </p>
    <p
      class="valueDay"
    >
      Em 30 dias: 
      <b>
        R$ 12,00
      </b>
    </p>
    <p
      class="valueDay"
    >
      Em 90 dias: 
      <b>
        R$ 13,00
      </b>
    </p>
  </aside>
</div>
`;

exports[`ValueToReceive should render correctly without 15 days 1`] = `
<div>
  <aside
    class="aside"
  >
    <p
      class="title"
    >
      Você receberá
    </p>
    <hr
      class="line"
    />
    <p
      class="valueDay"
    >
      Amanhã: 
      <b>
        R$ 10,00
      </b>
    </p>
    <p
      class="valueDay"
    >
      Em 30 dias: 
      <b>
        R$ 12,00
      </b>
    </p>
    <p
      class="valueDay"
    >
      Em 90 dias: 
      <b>
        R$ 13,00
      </b>
    </p>
  </aside>
</div>
`;

exports[`ValueToReceive should render correctly without 30 days 1`] = `
<div>
  <aside
    class="aside"
  >
    <p
      class="title"
    >
      Você receberá
    </p>
    <hr
      class="line"
    />
    <p
      class="valueDay"
    >
      Amanhã: 
      <b>
        R$ 10,00
      </b>
    </p>
    <p
      class="valueDay"
    >
      Em 15 dias: 
      <b>
        R$ 11,00
      </b>
    </p>
    <p
      class="valueDay"
    >
      Em 90 dias: 
      <b>
        R$ 13,00
      </b>
    </p>
  </aside>
</div>
`;

exports[`ValueToReceive should render correctly without 90 days 1`] = `
<div>
  <aside
    class="aside"
  >
    <p
      class="title"
    >
      Você receberá
    </p>
    <hr
      class="line"
    />
    <p
      class="valueDay"
    >
      Amanhã: 
      <b>
        R$ 10,00
      </b>
    </p>
    <p
      class="valueDay"
    >
      Em 15 dias: 
      <b>
        R$ 11,00
      </b>
    </p>
    <p
      class="valueDay"
    >
      Em 30 dias: 
      <b>
        R$ 12,00
      </b>
    </p>
  </aside>
</div>
`;

exports[`ValueToReceive should render correctly without tomorrow 1`] = `
<div>
  <aside
    class="aside"
  >
    <p
      class="title"
    >
      Você receberá
    </p>
    <hr
      class="line"
    />
    <p
      class="valueDay"
    >
      Em 15 dias: 
      <b>
        R$ 11,00
      </b>
    </p>
    <p
      class="valueDay"
    >
      Em 30 dias: 
      <b>
        R$ 12,00
      </b>
    </p>
    <p
      class="valueDay"
    >
      Em 90 dias: 
      <b>
        R$ 13,00
      </b>
    </p>
  </aside>
</div>
`;
